### ISSUE TITLE
[Question] How can I load the modified project back into VS2019 for editing?

### ISSUE THREAD

**Issue by shawty** [rr-wfm/MSBuild.Sdk.SqlProj#41]
Like many people trying to build dotnet core projects on linux, that include SQL server projects, I ended up here after some internet searching and research.

Like others, I'm editing in Visual Studio, checking into GitSVN then having JenkinsCI build my project on a linux target.

Before finding this project, this was my issue

![image](https://user-images.githubusercontent.com/1876454/91228631-78360f80-e720-11ea-8c3f-30c4489b1d88.png)

After following the instructions on your main page, and replacing my project file with that shown (The sql server project was created from the VS2019 template) I now find I can build the project from the command line using "dotnet build", but I can no longer load the project into Visual Studio 2019 for editing, when I try I just get the following:

![image](https://user-images.githubusercontent.com/1876454/91228851-d7941f80-e720-11ea-80bc-5124098b04aa.png)

How can I load the project into VS for editing, but still use your project to enable me to build my solution without the presence of the project breaking the build.

**Comment by ErikEJ** [comment]
Could you share your .csproj ?

**Comment by shawty** [comment]
Yea sure Erik, but there's nothing special in it, it's just the standard template one. (sqlproj) I think it is.

![image](https://user-images.githubusercontent.com/1876454/91288610-19f14700-e789-11ea-87ec-9c1ab349d940.png)

![image](https://user-images.githubusercontent.com/1876454/91288686-32f9f800-e789-11ea-9518-2813dee519bf.png)

![image](https://user-images.githubusercontent.com/1876454/91288728-3f7e5080-e789-11ea-8edb-8438a44429c5.png)

![image](https://user-images.githubusercontent.com/1876454/91288845-6046a600-e789-11ea-81f0-a51f7a1b4a68.png)

This ^^^^^ above, plus the attached ".sqlproj" file, is the original project files

[st_database.sqlproj.zip](https://github.com/rr-wfm/MSBuild.Sdk.SqlProj/files/5128985/st_database.sqlproj.zip)

I then replaced the sqlproj file created by the template with 

![image](https://user-images.githubusercontent.com/1876454/91289197-e95ddd00-e789-11ea-9460-540fe00f21dc.png)

as suggested in this projects readme.

That got me to the point where I can do this

![image](https://user-images.githubusercontent.com/1876454/91289293-1611f480-e78a-11ea-9451-9011a1750037.png)

but where reloading it into VS19, gives me this

![image](https://user-images.githubusercontent.com/1876454/91289372-2e820f00-e78a-11ea-84c6-39b5336d16ce.png)

**Comment by shawty** [comment]
I confess I did spend an hour or two last night chopping and changing bits of the project file around, but didn't get anywhere.

**Comment by jmezach** [comment]
@shawty The project file must have the `.csproj` extension. It will not work with `.sqlproj` because then Visual Studio will interpret it as an old style project, but since you've changed the contents of the project file to use SDK-style projects you get the error. I don't think there's currently a way we can support the `.sqlproj` extension.

**Comment by ErikEJ** [comment]
Maybe the documentation could be slightly clearer on that?

**Comment by shawty** [comment]
@jmezach ah ok.  So question?  It's not 100% clear from the read me, but how would I get an SQL server style project into my .NET core solution?

From scratch?

Basically, I would like to use the features of VS to help me edit, design and build the new tables/sprocs/fuctions etc, and maybe even generate a DACPAC for them (Although this is not a major requirement).

Then when checking into git, I still want my CI/CD to build the solution (I don't care too much if the SQL server stuff is built or not) but I don't want the SQL project to break my CI Build, which is what it's doing at present with just the normal standard SQL Server project files.

I ended up here, after several searches as this being the solution to that problem :-)

Sorry I'm rambling a bit :-)

In a nutshell, I want to be able to use VS19's SQL editing features, and be able to check in the SQL code along side all the CS code, but still have my CI service build the solution and produce the output on the Linux build server it's using.  Everything is Core based, and the build is performed using the standard "dotnet" cli tools.

**Comment by jmezach** [comment]
What I've heard people do is to have a "companion" `.csproj` next to the existing `.sqlproj`. I believe @ErikEJ has some experience with that as well. Basically you would create a new project (I believe it has to be in a separate folder as well) and then add the following to your project file:

```xml
<ItemGroup>
  <Content Include="../<sql-proj-folder>/**/*.sql" />
</ItemGroup>
```

You can then use the `.csproj` for CI and use the `.sqlproj` for editing.

**Comment by shawty** [comment]
@jmezach Thanks, I'll give that a try and let you know how I get on.

**Comment by shawty** [comment]
> 
> 
> Maybe the documentation could be slightly clearer on that?

Absolutley Erik.

Everything I've read that lead me to this point, suggests that your project is the solution, for not only my case, but TFS based builds that break with the same tooling errors too.

The best way to test it, is to build a regular .NET core web-api service, then add a new project to that solution, and make the project type be an SQL one.  Then use a linux machine set up with the standard .NET core tools, to perform a "dotnet build" from the cli, at solution level, so all the projects are built in one go.

You'll get something similar to my first image in the thread, and searching in those exceptions, will eventually lead you to here.

Some documentation changes, and/or possibly a page in the wiki telling people how to convert an SQL server project to build with the CLI tools in this case would be very helpful.

**Comment by ErikEJ** [comment]
I have a blog post here describing the "buddy" project scenario: https://erikej.github.io/efcore/2020/05/11/ssdt-dacpac-netcore.html

**Comment by ErikEJ** [comment]
Maybe a simple change to the docs:

Usage
The simplest usage is to create a new .csproj project file with the following contents:

**Comment by shawty** [comment]
Yep, that makes it clearer that you shouldn't replace the SQL Project.

I would also put the link to your blog post in there too.

At a minimum I would say

"The simplest usage is to create a new .csproj project file, alongside all the existing files, with the following contents:"

**Comment by jeffrosenberg** [comment]
Our usage may not be typical, but what we're doing on my team is committing only the .csproj file, but maintaining a script that can be used to generate the .sqlproj file on an adhoc basis. Because the .csproj file supports wildcards, it doesn't need to be changed with each commit like the .sqlproj file does. So most of the time, we rely on just the .csproj file and develop in a text editor. When we need to use SSDT, we generate a new .sqlproj file and open that in SSDT.

**Comment by shawty** [comment]
thanks @jeffrosenberg more fuel for the fire :-)

**Comment by shawty** [comment]
Still getting a bit stuck here.

Trying Erik's approach from his blog post, and dotnet core build is getting confused.

![image](https://user-images.githubusercontent.com/1876454/91311174-ae1fd600-e7aa-11ea-8751-dfba33d5aed1.png)

**Comment by shawty** [comment]
and if I try it at solution level:

![image](https://user-images.githubusercontent.com/1876454/91311544-1cfd2f00-e7ab-11ea-9a0e-5b0a5a35b7db.png)

**Comment by ErikEJ** [comment]
As @jmezach wrote, and as clearly stated in my blog post, do not have both the .sqlproj and .csproj in the same folder.

**Comment by shawty** [comment]
ah ok, I missed that bit... dunno why... ok, trying again.

**Comment by shawty** [comment]
Even with everything in 2 separate projects, the command line build still fails.

![image](https://user-images.githubusercontent.com/1876454/91315236-59cb2500-e7af-11ea-9c3e-2e10a7790a9e.png)

In VS19 all is ok : 

![image](https://user-images.githubusercontent.com/1876454/91315282-6b143180-e7af-11ea-87da-f2d39386aced.png)

But the command line has gone back to my very first error now, failing because it can't find the SSDT tools

**Comment by ErikEJ** [comment]
You need to exclude the .sqlproj from your build configuration.

**Comment by shawty** [comment]
Annnnnndddddd.... W0000000 H00000000 :-)

On a hunch, I turned off building of actual db project solution wide, and now everything works perfectly.

![image](https://user-images.githubusercontent.com/1876454/91316372-a3683f80-e7b0-11ea-866d-0f49c29617bf.png)

🎉🎉🎉🎉

**Comment by shawty** [comment]
> 
> 
> You need to exclude the .sqlproj from your build configuration.

Github didn't update the page until I posted my reply....

**Comment by shawty** [comment]
@ErikEJ if you don't object, I'm gonna blog this, and point folks both at your blog post, this project/issue and repeat the steps from your post with a bit extra detail.

Cheers
Shawty

**Comment by ErikEJ** [comment]
Excellent, the more noise the project makes, the better!

**Comment by shawty** [comment]
LOL... I won't copy it, but I will follow the same process, just using my own words.

**Comment by ErikEJ** [comment]
I have updadated my blog post.

**Comment by jmezach** [comment]
Thanks @ErikEJ for updating the docs. I think there isn't much more we can do here.

FYI: I've recorded a [dotnetFlix](https://www.dotnetflix.com) episode today about this project demoing it which also goes into converting an existing project, etc. Should be coming online soon.

**Comment by ErikEJ** [comment]
Looking forward to watching!

**Comment by shawty** [comment]
@jmezach kewl, I'll have to watch that.    I'll reference it from my blog post too, which I'll be writing later on this evening.

**Comment by shawty** [comment]
@jmezach by the way if your interested, I routinely do sessions for .NET channels and such like, so if you'd like me to do a video session on dotNetFlix, do feel free to reach out to me.  I'm one of the management team behind "LiDNUG (The Linked .NET Users Group on Linked-in)" (Lidnug.org) if you want to get in touch.

**Comment by shawty** [comment]
@ErikEJ and others - blogged : https://shawtyds.wordpress.com/2020/08/26/using-a-full-framework-sql-server-project-in-a-net-core-project-build/
